### PR TITLE
Stax porting

### PR DESCRIPTION
**Pull request by sgliner-ledger**

### CODE REVIEW

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -12,37 +12,6 @@ on:
...
-        uses: actions/checkout@v2
-      - name: Build
-        run: make BOLOS_SDK=$NANOX_SDK
-      - name: Upload app binary
-        uses: actions/upload-artifact@v2
-        with:
-          name: celo-app
-          path: bin
 
   job_scan_build:
```
Comment (xchapron-ledger) [inline .github/workflows/ci-workflow.yml:16]: Can you rather use the guideline enforcer ? https://github.com/LedgerHQ/app-boilerplate/blob/master/.github/workflows/guidelines_enforcer.yml
It part of it?

File: src/ui_common.h [hunk 2]
```diff
@@ -4,16 +4,38 @@
...
+void ui_approval_celo_relock_flow(void);
+void ui_approval_celo_create_account_flow(void);
+void ui_approval_celo_data_warning_gateway_tx_flow(void);
+void ui_approval_celo_gateway_tx_flow(void);
+void ui_approval_celo_data_warning_tx_flow(void);
+void ui_approval_celo_tx_flow(void);
 
-unsigned int io_seproxyhal_touch_data_ok(const bagl_element_t *e);
-unsigned int io_seproxyhal_touch_data_cancel(const bagl_element_t *e);
+unsigned int io_seproxyhal_touch_data_ok(void);
```
Comment (xchapron-ledger) [inline src/ui_common.h:31]: If you are brave enough, I thing those can be renamed...
`ui_callback_data_ok` or something like that would probably be better...

File: tests/python/test_sign_transaction.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+                "",                      #R
+                "",                      #S
+                ""                       #Done
+                ]
+            ):
+        navigator.navigate_and_compare(TESTS_ROOT_DIR, test_name, instructions)
+    response: bytes = get_async_response(backend)
+
+    assert (response.status == StatusCode.STATUS_OK)
+    sleep(0.1)
```
Comment (xchapron-ledger) [inline tests/python/test_sign_transaction.py:45]: Why those sleep?
Are they still necessary?
If not please remove them.

**Review by xchapron-ledger — APPROVED** [review]

File: tests/python/utils.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    return instructions
+
+
+def get_stax_review_instructions(num_screen_skip):
+    instructions = [NavIns(NavInsID.USE_CASE_CHOICE_CONFIRM)]
+
+    for i in range(num_screen_skip):
+        instructions.append(NavIns(NavInsID.USE_CASE_REVIEW_TAP))
+        instructions.append(NavIns(NavInsID.USE_CASE_REVIEW_PREVIOUS))
+        instructions.append(NavIns(NavInsID.USE_CASE_REVIEW_TAP))
```
[the remark this answers is not in the document]
Comment (sgliner-ledger) [reply tests/python/utils.py:20]: Hehe, alright :)

**Pull request merged**